### PR TITLE
[Asset] prefix asset name by slash if not found in manifest.json

### LINKED ISSUE #39084: [Asset] JsonManifestVersionStrategy doesn't deal with abstract url
**Symfony version(s) affected**: >=5.0

**Description**  
taking this simple `manifest.json` :
```json
{
  "main.css": "buil/css/app.b916426ea1d10021f3f17ce8031f93c2.css"
}
```
then generate an url for both `/main.css` and `main.css` : 
```php
$package = new Package(new  JsonManifestVersionStrategy(__DIR__.'/manifest.json'));

dump($package->getUrl('main.css'));
dump($package->getUrl('/main.css'));
```
We get as result : 
image

Even if `main.css` and `\main.css` represent the same file.

**How to reproduce**  

**Possible Solution**  

This is just a simple example that treats only one case : 
```diff
<?php

/*
 * This file is part of the Symfony package.
 *
 * (c) Fabien Potencier <fabien@symfony.com>
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */

namespace Symfony\Component\Asset\VersionStrategy;

/**
 * Reads the versioned path of an asset from a JSON manifest file.
 *
 * For example, the manifest file might look like this:
 *     {
 *         "main.js": "main.abc123.js",
 *         "css/styles.css": "css/styles.555abc.css"
 *     }
 *
 * You could then ask for the version of "main.js" or "css/styles.css".
 */
class JsonManifestVersionStrategy implements VersionStrategyInterface
{
    private $manifestPath;
    private $manifestData;

    /**
     * @param string $manifestPath Absolute path to the manifest file
     */
    public function __construct(string $manifestPath)
    {
        $this->manifestPath = $manifestPath;
    }

    /**
     * With a manifest, we don't really know or care about what
     * the version is. Instead, this returns the path to the
     * versioned file.
     */
    public function getVersion(string $path)
    {
        return $this->applyVersion($path);
    }

    public function applyVersion(string $path)
    {
        return $this->getManifestPath($path) ?: $path;
    }

    private function getManifestPath(string $path): ?string
    {
        if (null === $this->manifestData) {
            if (!file_exists($this->manifestPath)) {
                throw new \RuntimeException(sprintf('Asset manifest file "%s" does not exist.', $this->manifestPath));
            }

            $this->manifestData = json_decode(file_get_contents($this->manifestPath), true);
            if (0 < json_last_error()) {
                throw new \RuntimeException(sprintf('Error parsing JSON from asset manifest file "%s": ', $this->manifestPath).json_last_error_msg());
            }
        }

--        return isset($this->manifestData[$path]) ? $this->manifestData[$path] : null;
++        return $this->manifestData[$path] ?? $this->getPossiblePath($path);
    }
++
++    private function getPossiblePath(string $path): ?string
++    {
++        return $this->manifestData[$this->generatePossiblePath($path)] ?? null;
++    }
++
++    private function generatePossiblePath(string $path): string
++    {
++        return 0 === strpos($path, '/') ? ltrim($path, '/') : '/'.$path;
++    }
}
```

Result for the previous example : 
image

**Additional context**

### PR DESCRIPTION
**Pull request by ybenhssaien**
| Q             | A
| ------------- | ---
| Branch?       | 5.x
| Bug fix?      | yes
| New feature?  | yes
| Deprecations? | no
| Tickets       | Fix #39084 
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

# Modifs
- [x] unit tests
- [x] CHANGELOG

# Description
get the generated asset calling `main.css` or `/main.css` if not found
```json
{
  "main.css": "buil/css/app.b916426ea1d10021f3f17ce8031f93c2.css"
}
```
```php
$package = new Package(new  JsonManifestVersionStrategy(__DIR__.'/manifest.json'));

dump($package->getUrl('main.css'));
dump($package->getUrl('/main.css'));
```

### CODE REVIEW

File: src/Symfony/Component/Asset/VersionStrategy/JsonManifestVersionStrategy.php [hunk 1]
```diff
@@ -63,6 +63,16 @@ private function getManifestPath(string $path): ?string
             }
         }
 
-        return isset($this->manifestData[$path]) ? $this->manifestData[$path] : null;
+        return $this->manifestData[$path] ?? $this->getPossiblePath($path);
```
Comment (jderusse) [inline src/Symfony/Component/Asset/VersionStrategy/JsonManifestVersionStrategy.php]: content of the 2 methods can be inlined here IMHO.

**Review by jderusse — COMMENTED** [review]
I'm not convinced by the benefit of this behavior.

If the asset is FOO, people should use `asset(FOO)` instead of `asset(/FOO)`.

We are adding complexity in Symfony to handle developer's mistake.

**Comment by ybenhssaien** [comment]
> I'm not convinced by the benefit of this behavior.
> 
> If the asset is FOO, people should use `asset(FOO)` instead of `asset(/FOO)`.
> 
> We are adding complexity in Symfony to handle developer's mistake.

Sometimes is not the developer's mistake (here is an old issue reports a case #36234 => maybe a migration or configuration changes), in the most cases the method should return `$this->manifestData[$path]`, otherwise instead of returning `null` we may give a second chance to handle the different representations of the same name, finally `/main.css` and `main.css` are the same, so logically the component may handle this misconfiguration I guess, what to do you think?

File: src/Symfony/Component/Asset/VersionStrategy/JsonManifestVersionStrategy.php (hunk 1 above)
Comment (ybenhssaien) [reply src/Symfony/Component/Asset/VersionStrategy/JsonManifestVersionStrategy.php]: 👍 yes now seems better thank you ^^

**Comment by nicolas-grekas** [comment]
I'm :-1: here. This adds complexity and is unexpected.

**Comment by stof** [comment]
`asset('main.css')` and `asset('/main.css')` are also producing a different behavior in other strategies than the manifest strategy. So that's also a -1 from me.

**Comment by nicolas-grekas** [comment]
Thank you for proposing @ybenhssaien

**Pull request closed without merging**